third_party/terraform/resources/resource_bigtable_instance.go

```diff
@@ -369,7 +378,19 @@ func expandBigtableClusters(clusters []interface{}, instanceID string) []bigtabl
 			StorageType: storageType,
 		})
 	}
-	return results
+	return results, nil
+}
+
+// getBigtableZone reads the "zone" value from the given resource data and falls back
+// to provider's value if not given.  If neither is provided, returns an error.
+func getBigtableZone(z string, config *Config) (string, error) {
+	if z == "" {
+		if config.Zone != "" {
+			return config.Zone, nil
+		}
+		return "", fmt.Errorf("cannot determine zone: set in cluster.0.zone, or set provider-level zone")
+	}
+	return GetResourceNameFromSelfLink(z), nil
```

Comment:
it seems like if we get here, the expected value would be a zone name like `us-central1-b`, rather than a self-link. Is that correct? I think this would still work, but it should be equivalent to returning `z` directly.

Reply:
I just copied and tweaked the types for getZone function that is everywhere.

Reply:
🤔  yeah, it looks like getZone does the same. That's also weird 😂  but let's just go with it I guess.